src/pages/Products.js

```diff
@@ -0,0 +1,332 @@
+              "Content-Type": "application/json",
+            },
+          }
+        );
+        const data = response.data;
+        if (data && data.list) {
+          const sortedData = data.list.sort(
+            (a, b) => b.favoriteCount - a.favoriteCount
+          );
+          setBestProducts(sortedData.slice(0, 4));
+
+          const sortedProducts = data.list.slice().sort((a, b) => {
+            if (order === "recent") {
+              return new Date(b.createdAt) - new Date(a.createdAt);
+            } else if (order === "favorite") {
+              return b.favoriteCount - a.favoriteCount;
+            }
+            return 0;
+          });
+          setProducts(sortedProducts);
+        } else {
+          alert("에러가 발생했습니다.");
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    fetchProducts();
+  }, [order]);
+
+  const handleOrderChange = (selectedOrder) => {
+    setOrder(selectedOrder);
+    setIsOpen(false); // Close dropdown after selecting an option
+  };
+
+  const handleProductClick = (productId) => {
+    navigate(`/items/${productId}`);
+  };
+
+  return (
+    <>
+      <Header />
+      <Container>
+        <BestSectionTitle>베스트 상품</BestSectionTitle>
+        <BestProductContainer>
+          {bestProducts &&
+            bestProducts.map((product) => (
+              <BestProductItem
+                key={product.id}
+                onClick={() => handleProductClick(product.id)}
+              >
+                <BestProductImage src={product.images} alt={product.name} />
+                <ProductName>{product.name}</ProductName>
+                <ProductPrice>{product.price}원</ProductPrice>
+                <ProductLikes>
+                  <Heart src={heart}></Heart>
+                  {product.favoriteCount}
+                </ProductLikes>
+              </BestProductItem>
+            ))}
+        </BestProductContainer>
+        <div style={{ display: "flex", margin: "20px 0" }}>
+          <SectionTitle>전체 상품</SectionTitle>
+          <Tools>
+            <Search>
+              <img
+                src={search}
+                alt="검색"
+                style={{
+                  position: "relative",
+                  top: "10px",
+                  left: "40px",
+                  width: "24px",
+                  height: "24px",
+                }}
+              />
+              <SearchInput
+                type="text"
+                placeholder="검색할 상품을 입력해주세요"
+              ></SearchInput>
+            </Search>
+            <ProductRegister onClick={() => navigate("/additem")}>
+              상품 등록하기
+            </ProductRegister>
+            <CustomSelect>
+              <SelectButton onClick={() => setIsOpen(!isOpen)}>
+                <Order>{order === "recent" ? "최신순" : "좋아요순"}</Order>
+                <Arrow src={arrowDown} alt="드롭다운 화살표" />
+              </SelectButton>
+              <OptionsContainer isOpen={isOpen}>
```

Comment:
개발빌드로 서버구동시(npm start), 아래 에러가 발생합니다
```
Warning: React does not recognize the `isOpen` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `isopen` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
```

[transient prop](https://styled-components.com/docs/api#transient-props) 으로 명시해주세요
```suggestion
              <OptionsContainer $isOpen={isOpen}>
```